crates/chia-consensus/src/gen/build_compressed_block.rs

```diff
@@ -0,0 +1,417 @@
+                7_000_000,
+                0,
+                &TEST_CONSTANTS,
+            )
+            .expect("run_spendbundle")
+            .0;
+
+            if conds
+                .spends
+                .iter()
+                .any(|s| seen_spends.contains(&*s.coin_id))
+            {
+                // We can't have conflicting spend bundles, since we combine
+                // them randomly. In this case two spend bundles spend the same
+                // coin
+                panic!(
+                    "conflict in {}",
+                    file.file_name().unwrap().to_str().unwrap()
+                );
+            }
+            if conds.spends.iter().any(|s| {
+                s.create_coin.iter().any(|c| {
+                    seen_spends.contains(&Coin::new(*s.coin_id, c.puzzle_hash, c.amount).coin_id())
+                })
+            }) {
+                // We can't have conflicting spend bundles, since we combine
+                // them randomly. In this case one spend bundle spends the coin
+                // created by another. This is probably OK in most cases, but
+                // not in the general case. We have restrictions on ephemeral
+                // spends (they cannot have relative time-lock conditions).
+                // Since the combination is random, we may end up with an
+                // invalid block.
+                panic!(
+                    "conflict in {}",
+                    file.file_name().unwrap().to_str().unwrap()
+                );
+            }
+            for spend in &conds.spends {
+                seen_spends.insert(*spend.coin_id);
+                for coin in &spend.create_coin {
+                    seen_spends
+                        .insert(Coin::new(*spend.coin_id, coin.puzzle_hash, coin.amount).coin_id());
+                }
+            }
+
+            // cost is supposed to not include byte-cost, so we have to subtract
+            // it here
+            let cost = conds.cost
+                - (calculate_generator_length(&bundle.coin_spends) as u64 - 2)
+                    * TEST_CONSTANTS.cost_per_byte;
+
+            let mut conds = OwnedSpendBundleConditions::from(&a, conds);
+            // when running a block in consensus mode, we don't bother
+            // establishing whether a spend is eligible for dedup or not.
+            // So, to compare with the generator output later, we need to clear
+            // this field
+            for s in &mut conds.spends {
+                s.flags = 0;
+                // when parsing conditions, create coin conditions are stored in
+                // a hash set to cheaply check for double spends. This means the
+                // order of this collection is not deterministic. In order to
+                // compare to the generator output later, we need to sort both.
+                s.create_coin.sort();
+            }
+            all_bundles.push(Box::new((bundle, cost, conds)));
```

Comment:
Out of curiosity, why did you need a box for these vector elements?

Reply:
they are shared between the threads below